geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/clustering/ClusteringUtils.java

```diff
@@ -0,0 +1,409 @@
+package mil.nga.giat.geowave.analytics.clustering;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import mil.nga.giat.geowave.accumulo.AccumuloOperations;
+import mil.nga.giat.geowave.accumulo.BasicAccumuloOperations;
+import mil.nga.giat.geowave.accumulo.mapreduce.input.GeoWaveInputConfigurator;
+import mil.nga.giat.geowave.accumulo.metadata.AccumuloAdapterStore;
+import mil.nga.giat.geowave.accumulo.metadata.AccumuloIndexStore;
+import mil.nga.giat.geowave.analytics.tools.AnalyticFeature;
+import mil.nga.giat.geowave.analytics.tools.dbops.BasicAccumuloOperationsBuilder;
+import mil.nga.giat.geowave.analytics.tools.dbops.DirectBasicAccumuloOperationsBuilder;
+import mil.nga.giat.geowave.index.ByteArrayId;
+import mil.nga.giat.geowave.index.ByteArrayRange;
+import mil.nga.giat.geowave.index.NumericIndexStrategy;
+import mil.nga.giat.geowave.index.NumericIndexStrategyFactory.DataType;
+import mil.nga.giat.geowave.index.sfc.data.MultiDimensionalNumericData;
+import mil.nga.giat.geowave.index.sfc.data.NumericData;
+import mil.nga.giat.geowave.index.sfc.data.NumericRange;
+import mil.nga.giat.geowave.store.adapter.DataAdapter;
+import mil.nga.giat.geowave.store.index.CustomIdIndex;
+import mil.nga.giat.geowave.store.index.DimensionalityType;
+import mil.nga.giat.geowave.store.index.Index;
+import mil.nga.giat.geowave.store.index.IndexType;
+import mil.nga.giat.geowave.store.query.DistributableQuery;
+import mil.nga.giat.geowave.store.query.SpatialQuery;
+import mil.nga.giat.geowave.vector.adapter.FeatureDataAdapter;
+
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.accumulo.core.client.IteratorSetting;
+import org.apache.accumulo.core.iterators.user.RegExFilter;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.geom.LinearRing;
+import com.vividsolutions.jts.geom.Polygon;
+import com.vividsolutions.jts.geom.PrecisionModel;
+import com.vividsolutions.jts.geom.impl.CoordinateArraySequence;
+import com.vividsolutions.jts.io.ParseException;
+import com.vividsolutions.jts.io.WKTReader;
+
+public class ClusteringUtils
+{
+
+	final static Logger LOGGER = LoggerFactory.getLogger(ClusteringUtils.class);
+	private static BasicAccumuloOperationsBuilder basicAccumuloOperationsBuilder = new DirectBasicAccumuloOperationsBuilder();
```

Comment:
complete nit, but figured I'd mention it: not quite a "builder" (ie. it doesn't take each parameter step by step), a bit more like a factory, also 'final' and the java constant naming convention would be good for code consistency between developers (ie. ALL_CAPS)